Fix issue #304: orphaned VLAN interfaces

markgoddard opened the pull request:
Skydive does not show the relationship between Linux VLAN interfaces and
their parent interfaces. This is because the netlink topology probe
looks up the Vlan metadata field as a String, when adding the
child/parent relationship, but in fact Vlan is an integer. This change
fixes this issue by using the more generic GetField() function to look
up the Vlan metadata field.

safchain commented on topology/probes/netlink.go:
```diff
@@ -116,7 +116,7 @@ func (u *NetNsNetLinkProbe) handleIntfIsChild(intf *graph.Node, link netlink.Lin
 
 	if link.Attrs().ParentIndex != 0 {
-		if _, err := intf.GetFieldString("Vlan"); err == nil {
+		if _, found := intf.GetField("Vlan"); found {
```
Maybe that would be better to fix the type of the vlan field to int64 line 375, so that we can use the GetFieldInt64 function here. This will help for Gremlin request to have the proper type.

markgoddard replied:
Done

markgoddard commented:
Thanks for the review. It's my first commit to skydive, so not familiar with the conventions. I've updated the commit message as requested. The Vlan field is now converted to an int64 as recommended.

safchain commented:
@markgoddard Thanks for updating your PR!
